src/v2/view-builder/views/webauthn/EnrollWebauthnView.js

```diff
@@ -0,0 +1,126 @@
+    };
+  });
+}
+
+const Body = BaseForm.extend({
+  title: loc('oie.enroll.webauthn.title', 'login'),
+  className: 'oie-webauthn',
+  modelEvents: {
+    'error': '_stopEnrollment',
+  },
+  getUISchema () {
+    const schema = [];
+    const activationData = this.options.appState.get('currentAuthenticator').contextualData.activationData;
+    // Returning custom array so no input fields are displayed for webauthn
+    if (webauthn.isNewApiAvailable()) {
+      schema.push({
+        View: `
+          <p class="idx-webauthn-enroll-text">{{i18n code="oie.enroll.webauthn.instructions" bundle="login"}}</p>`,
+      });
+
+      if (BrowserFeatures.isEdge()) {
+        schema.push({
+          View: `
+            <p class="idx-webauthn-enroll-text-edge">
+              {{i18n code="oie.enroll.webauthn.instructions.edge" bundle="login"}}
+            </p>`,
+        });
+      }
+
+      if (activationData.authenticatorSelection.userVerification === 'required') {
+        schema.push({
+          View: createCallout({
+            className: 'uv-required-callout',
+            size: 'slim',
+            type: 'warning',
+            subtitle: loc('oie.enroll.webauthn.uv.required.instructions', 'login'),
+          }),
+        });
+      }
+
+      schema.push({
+        View: '<div data-se="webauthn-waiting" class="okta-waiting-spinner"></div>'
+      });
+      schema.push({
+        View: createButton({
+          className: 'webauthn-setup button button-primary button-wide',
+          title: loc('oie.enroll.webauthn.save', 'login'),
+          click: () => {
+            this.triggerWebauthnPrompt();
+          }
+        }),
+      });
+    } else {
+      schema.push({
+        View: createCallout({
+          className: 'webauthn-not-supported',
+          type: 'error',
+          subtitle: loc('oie.enroll.webauthn.error.not.supported', 'login'),
+        }),
+      });
+    }
+    return schema;
+  },
+  triggerWebauthnPrompt () {
+    this.$el.find('.o-form-error-container').empty();
+    this._startEnrollment();
+    const activationData = this.options.appState.get('currentAuthenticator').contextualData.activationData;
+    if (webauthn.isNewApiAvailable()) {
+      var options = _.extend({}, activationData, {
+        challenge: CryptoUtil.strToBin(activationData.challenge),
+        user: {
+          id: CryptoUtil.strToBin(activationData.user.id),
+          name: activationData.user.name,
+          displayName: activationData.user.displayName
+        },
+        excludeCredentials: getExcludeCredentials(activationData.excludeCredentials)
+      });
+      this.webauthnAbortController = new AbortController();
+      navigator.credentials.create({
+        publicKey: options,
+        signal: this.webauthnAbortController.signal
+      }).then((newCredential) => {
+        this.model.set({
+          credentials : {
+            clientData: CryptoUtil.binToStr(newCredential.response.clientDataJSON),
```

Comment:
Just for curiosity, does it make sense to change name in API so that we'd be able avoid hard code names.
- `clientData -> clientDataJSON`
- `assestation -> attestationObject`

Reply:
Might be a bit late. @mpeng-okta ^

Reply:
@haishengwu-okta I don't quite follow what is meant by "avoid hard code names". Would you mind elaborating -- are these not also hard-coded? Thanks!

Reply:
> "avoid hard code names"

The UI has to know map `response .clientDataJSON` to `clientData` that API expects.
If API sends UI name as `clientDataJSON`, the UI can do  `newCredential.response[ionFIeldName]` without worrying what's the actual ionFieldName.

Reply:
Actually, I'm not sure about the name change, because the `clientDataJSON` is as named, but we're not actually sending `clientDataJSON` here -- we're sending an object that's the base64-encoded version of `clientDataJSON`, and calling it `clientData`.

I think I would therefore prefer to name it specifically as we're doing here, since it's not actually the JSON received from the WebAuthn API. Let me know if that makes sense to you as well.

Reply:
oh, I see. Thanks!

Reply:
It is not only that the clientData is and assestation is wrapped under credentials too. so at the very least SIW needs to put them into credentials. and that needs to be hardcoded.